monai/inferers/inferer.py

```diff
@@ -64,6 +67,166 @@ def __call__(self, inputs: torch.Tensor, network: Callable[..., torch.Tensor], *
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
 
+class PatchInferer(Inferer):
+    """
+    SimpleInferer is the normal inference method that run model forward() directly.
+
+    Args:
+        splitter: a `Splitter` object that split the inputs into patches. Defaults to None.
+            If not provided or None, the inputs are considered to be already splitted into patches.
+        merger: a `Merger` object that merges patch outputs. Defaults to AvgMerger.
+        pre_processor: a callable that process patches before the being fed to the network. Defaults to None.
+        post_processor: a callable that process the output of the network. Defaults to None.
+        output_keys: if the network output is a dictionary, this defines the keys of the output dictionary to use.
+            Defaults to None, where all the keys are taken if output is a dictionary.
+    """
+
+    def __init__(
+        self,
+        splitter: Splitter,
+        merger: Merger | Sequence[Merger] | None = None,
+        batch_size: int = 1,
+        pre_processor: Callable | None = None,
+        post_processor: Callable | None = None,
+        patch_filter_fn: Callable | None = None,
+        output_keys: Sequence | None = None,
+    ) -> None:
+        Inferer.__init__(self)
+
+        # splitter
+        if not callable(splitter):
+            raise TypeError(f"'splitter' should be a callable object, {type(splitter)} is given.")
+        self.splitter = splitter
+
+        # mergers
+        self.mergers = (AvgMerger(),) if merger is None else ensure_tuple(merger)
+        for m in self.mergers:
+            if not isinstance(m, Merger):
+                raise TypeError(f"'merger' should be a `Merger` object, {type(m)} is given.")
+
+        # pre-processor (process patch before the network)
+        if pre_processor is not None and not callable(pre_processor):
+            raise TypeError(
+                f"'pre_processor' should be a callable object, not None and {type(pre_processor)} is given."
+            )
+        self.pre_processor = pre_processor
+
+        # post-processor (process the output of the network)
+        if post_processor is not None and not callable(post_processor):
+            raise TypeError(f"'post_processor' should be a callable object, {type(post_processor)} is given.")
+        self.post_processor = post_processor
+
+        # function to filter patches for inference
+        if patch_filter_fn is not None and not callable(patch_filter_fn):
+            raise TypeError(f"'patch_filter_fn' should be a callable object, {type(patch_filter_fn)} is given.")
+        self.patch_filter_fn = patch_filter_fn
+
+        # batch size for patches
+        self.batch_size = batch_size
+
+        # model output keys
+        self.output_keys = output_keys
+
+        # check if merger is initialized
+        self.is_merger_initialized = False
```

Comment:
This flag may cause the thread-safe issue, please double-check the logic.

Thanks.